CHANGES/2422.bugfix

```diff
@@ -0,0 +1,3 @@
+Taught advisory-conflict-resolution to handle just-EVR-differences in incoming advisory's
+package-list. This solves the case of repositories that update advisories to always have
+the newest versions of RPMs (looking at you, EPEL...).
```

Comment:
I'm interpreting this to mean that the RPM NEVRAs are changing from push to push without properly updating the other metadata.  Is that a correct assessment?

Would you argue that the behavior is erroneous?

Reply:
> I'm interpreting this to mean that the RPM NEVRAs are changing from push to push without properly updating the other metadata. Is that a correct assessment?

Updated-date and/or version changes. Package-list has the same Names, but newer(est) EVR.

> Would you argue that the behavior is erroneous?

Well, *I* would argue that errata are "a record of what was released into a repo, when, and why", and should therefore be **invariant** once released. That does not, alas, match the reality of the RPM ecosystem. And since none of this is documented in a shall/must/must-not format anywhere - the best we can do is handle the reality "on the ground" as best we can.

Side-effects of this particular bit of weirdness means that you can install an erratum, FOO, that provides bar-1.2.2-1.rpm, on a given machine, and then two weeks later ask "has FOO been installed here" - and the answer is "no", because **now** FOO says it needs bar-1.2.3-4. I don't like this behavior. But it exists "in the wild", and rejecting it breaks Pulp3 users.